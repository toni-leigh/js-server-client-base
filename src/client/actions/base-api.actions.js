const handleGetResponse = ({ json, responseConstant }) => ({
  type: responseConstant,
  data: json
});

export const buildGetAction = ({
  getUrl,
  responseConstant
}) => () => dispatch => {
  return fetch(getUrl)
    .then(response => response.json())
    .then(json => dispatch(handleGetResponse({ json, responseConstant })));
};