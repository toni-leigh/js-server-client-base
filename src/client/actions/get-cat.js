const GET_CAT = 'GET_CAT';
const RECEIVE_CAT = 'RECEIVE_CAT';

const getCat = () => dispatch => {
  return fetch('https://api.thecatapi.com/v1/images/search?size=full')
    .then(response => response.json())
    .then(json => dispatch(receiveCat(json)));
};

const receiveCat = cat => ({
  type: RECEIVE_CAT,
  payload: cat
});

export {
  getCat,
  GET_CAT,
  receiveCat,
  RECEIVE_CAT
};
