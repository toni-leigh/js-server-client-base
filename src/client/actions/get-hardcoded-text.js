const GET_HARDCODED_TEXT = 'GET_HARDCODED_TEXT';
const HARDCODED_TEXT_RECEIVED = 'HARDCODED_TEXT_RECEIVED';

const getHardcodedTextAction = () => dispatch => {
  return fetch('http://localhost:7555/hardcoded-text')
    .then(response => response.json())
    .then(json => dispatch(hardcodedTextReceived(json)));
};

const hardcodedTextReceived = hardcodedText => ({
  type: HARDCODED_TEXT_RECEIVED,
  data: { hardcodedText }
});

export {
  getHardcodedTextAction,
  GET_HARDCODED_TEXT,
  hardcodedTextReceived,
  HARDCODED_TEXT_RECEIVED
};
