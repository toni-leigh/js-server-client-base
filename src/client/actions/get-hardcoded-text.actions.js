import { buildGetAction } from './base-api.actions.js';

const HARDCODED_TEXT_RECEIVED = 'HARDCODED_TEXT_RECEIVED';

const getHardcodedTextAction = buildGetAction({
  getUrl: 'http://localhost:7555/hardcoded-text',
  responseConstant: HARDCODED_TEXT_RECEIVED
});

export {
  getHardcodedTextAction,
  HARDCODED_TEXT_RECEIVED
};