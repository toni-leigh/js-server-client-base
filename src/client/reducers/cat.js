import { fromJS } from 'immutable';

import { CAT_RECEIVED } from '../actions/get-cat';

const initialState = fromJS({ imageSrc: '' })

const cat = (state = initialState, action) => {
  switch (action.type) {
    case CAT_RECEIVED:
      return state.set('imageSrc', action.data[0].url);
    default:
      return state;
  }
}

export default cat;