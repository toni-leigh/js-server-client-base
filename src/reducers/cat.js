import { fromJS } from 'immutable';

import { RECEIVE_CAT } from '../actions/get-cat';

const initialState = fromJS({ imageSrc: '' })

const cat = (state = initialState, action) => {
  switch (action.type) {
    case RECEIVE_CAT:
      return state.set('imageSrc', action.payload[0].url);
    default:
      return state;
  }
}

export default cat;