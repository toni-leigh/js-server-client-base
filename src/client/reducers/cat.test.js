import { fromJS } from 'immutable';

import cat from './cat';
import { CAT_RECEIVED } from '../actions/get-cat';

test('cat reducer without state uses default', () => {
  expect(cat(undefined, { type: 'STUBBED' })).toMatchSnapshot();
});

test('CAT_RECEIVED', () => {
  const state = fromJS({ imageSrc: 'foo' });
  const action = { type: CAT_RECEIVED, data: [{ url: 'bar' }] };
  expect(cat(state, action)).toMatchSnapshot();
});