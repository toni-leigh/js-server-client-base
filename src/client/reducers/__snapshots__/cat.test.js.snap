// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RECEIVE_CAT 1`] = `
Immutable.Map {
  "imageSrc": "bar",
}
`;

exports[`cat reducer without state uses default 1`] = `
Immutable.Map {
  "imageSrc": "",
}
`;
